File: android/app/src/main/java/com/rafaeltmbr/stopwatch/infra/presentation/navigation/StackNavigator.kt
package com.rafaeltmbr.stopwatch.infra.presentation.navigation

import kotlinx.coroutines.flow.StateFlow

interface StackNavigator {
    sealed class Screen(val name: String) {
        data object Home : Screen("Home")
        data object Laps : Screen("Laps")
    }

    val stack: StateFlow<List<Screen>>
    suspend fun push(screen: Screen)
    suspend fun pop()
}

File: ios/Stopwatch/platform/presentation/navigation/StackNavigator.swift
import Foundation

enum StackNavigatorScreen {
    case home, laps
}

protocol StackNavigator{
    associatedtype Events: EventSubscriber where Events.Event == [StackNavigatorScreen]
    
    var stack: [StackNavigatorScreen] { get }
    var events: Events { get }
    
    func push(_ screen: StackNavigatorScreen)
    func pop()
}
